compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -584,50 +602,8 @@ fn find_opaque_ty_constraints(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Ty<'_> {
                 // FIXME(oli-obk): trace the actual span from inference to improve errors.
                 let span = self.tcx.def_span(def_id);
 
-                // HACK(eddyb) this check shouldn't be needed, as `wfcheck`
```

Comment:
Since the `mir_borrowck` call above will invoke the checking logic, duplicating it here with delay_span_bug adds no additional safety net